### ISSUE TITLE
crawl should do breadth first traverse

### ISSUE THREAD

**Issue by goto-bus-stop** [goto-bus-stop/scope-analyzer#2]
```js
function a () {
  function c( ){
    b () // reference is not detected
  }
}
function b () {}
```

**Comment by goto-bus-stop** [comment]
2.0.3 fixed this in a different way (does two walks)
